Scripts/Runtime/Data/Collections/DeferredNativeArray.cs

```diff
@@ -446,6 +446,21 @@ public unsafe NativeArray<T> AsDeferredJobArray()
             buffer += 1;
             NativeArray<T> array = NativeArrayUnsafeUtility.ConvertExistingDataToNativeArray<T>(buffer, 0, Allocator.Invalid);
 
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+            NativeArrayUnsafeUtility.SetAtomicSafetyHandle(ref array, m_Safety);
+#endif
+
+            return array;
+        }
+
+        public unsafe NativeArray<T> AsArray()
+        {
+            //This whole function taken from NativeList.AsDeferredJobArray
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+            AtomicSafetyHandle.CheckExistsAndThrow(m_Safety);
+#endif
+            NativeArray<T> array = NativeArrayUnsafeUtility.ConvertExistingDataToNativeArray<T>(m_BufferInfo->Buffer, m_BufferInfo->Length, Allocator.Invalid);
```

Comment:
Should this be `Allocator.Invalid` or whatever the buffer's allocator is?

Reply:
I suppose so. Fixed!